src/physics/physicsmodel.cxx

```diff
@@ -97,23 +97,17 @@ int PhysicsModel::postInit(bool restarting) {
   // Second argument specifies no time history
   solver->outputVars(restart, false);
 
-  std::string restart_dir;  ///< Directory for restart files
-  std::string dump_ext, restart_ext;  ///< Dump, Restart file extension
-  
-  Options *options = Options::getRoot();
-  if (options->isSet("restartdir")) {
-    // Solver-specific restart directory
-    options->get("restartdir", restart_dir, "data");
-  } else {
-    // Use the root data directory
-    options->get("datadir", restart_dir, "data");
-  }
-  /// Get restart file extension
-  const auto default_dump_format = bout::build::has_netcdf ? "nc" : "h5";
-  options->get("dump_format", dump_ext, default_dump_format);
-  options->get("restart_format", restart_ext, dump_ext);
+  auto& options = Options::root();
+
+  const std::string restart_dir = options["restartdir"]
+                                      .doc("Directory for restart files")
+                                      .withDefault(options["datadir"]);
+
+  const std::string restart_ext = options["restart_format"]
+                                      .doc("Restart file extension")
+                                      .withDefault(options["dump_format"]);
```

Comment:
Is it worth noting that we're assuming that `options["dump_format"]` has been set?

Reply:
Yes, good idea. It's only an issue if `BoutInitialise` isn't called, which is pretty rare